### DISCUSSION TITLE
Using Provisioning Engine with PnP.Core

### DISCUSSION THREAD

**Discussion by derhallim** [pnp/pnpcore#296] [Q&A] [answered]
#### Environment
[x] Office 365 / SharePoint Online
[ ] SharePoint 2016
[ ] SharePoint 2013

I have a .NET Core 3.1 console application. Will we be able to use PnP.Core to read and apply a provisioning template (.xml that was created with PnP Core PowerShell) using our console app? What's written [here](https://docs.microsoft.com/en-us/sharepoint/dev/solution-guidance/pnp-provisioning-engine-and-the-core-library) works with the old library (OfficeDevPnP.Core) that's marked as archived. Can we do the same thing with the new library (Calling ApplyProvisioningTemplate) or do we need to use the old one for provisioning for now?

@jansenbe , @ypcode , @pkbullock , @PaoloPia, @JarbasHorst

**Comment by jansenbe** [comment, marked as answer]
@derhallim : for now you should use PnP Framework if you want to use the provisioning features, we however plan to rewrite the provisioning and integrate this in PnP Core SDK.

**Reply by derhallim** [reply]
@jansenbe Thanks! I'll have a look at it and let you know if I have any issues.

**Reply by derhallim** [reply]
@jansenbe So I added the PnP.Framework and Microsoft.Extensions.Hosting packages. However am following that [doc ](https://docs.microsoft.com/en-us/sharepoint/dev/solution-guidance/provisioning-console-application-sample)along with the [Github repo](https://github.com/pnp/PnP/tree/master/Samples/Provisioning.Framework.Console/Provisioning.Framework.Console) it's giving me an error regarding

>  "SharePointOnlineCredentials" The type or namespace could not be found

Which package do I need to reference to include it?

Currently this is what I have imported: 

```
using System;

using System.Threading.Tasks;

using System.Security;

using PnP.Framework.Provisioning.Model;

using Microsoft.SharePoint.Client;

using System.Threading;

using PnP.Framework.Provisioning.ObjectHandlers;

using PnP.Framework.Provisioning.Connectors;

using PnP.Framework.Provisioning.Providers.Xml;

```

**Reply by derhallim** [reply]
OK that's because we can't use username/password with the new model.
